app/objects/c_ability.py

```diff
@@ -140,25 +91,26 @@ async def which_plugin(self):
                 return plugin
         return None
 
-    def replace_cleanup(self, encoded_cmd, payload):
-        decoded_cmd = b64decode(encoded_cmd).decode('utf-8', errors='ignore').replace('\n', '')
-        decoded_cmd = decoded_cmd.replace(self.RESERVED['payload'], payload)
-        return decoded_cmd
+    def find_executor(self, platform, name):
+        for executor in self.executors:
+            if executor.platform == platform and executor.name == name:
+                return executor
+        return None
+
+    def find_executors(self, platform, names):
```

Comment:
Need to keep ordering

Reply:
a `seen name` set might be better

Reply:
^^^ that `seen` for `names` approach would be nice because it means you wouldn't incur the cost of searching for an executor when it's already been processed and you would process the names in the order they are received.